src/transactions.rs

```diff
@@ -286,7 +291,13 @@ impl<L: Ledger> Transactions<L> {
         txn: &Transaction<L>,
     ) -> Result<(), KeystoreError<L>> {
         self.store.store(uid, txn)?;
-        self.reload();
+        if let Some(timeout) = txn.timeout() {
+            self.expiring_txns.insert((timeout, txn.uid().clone()));
```

Comment:
If `txn.timeout()` was already set, this still adds an `Add` action to the changelog, causing the timeout to be removed from the index on a call to `revert`.